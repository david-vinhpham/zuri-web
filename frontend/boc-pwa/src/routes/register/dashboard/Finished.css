.finished {
  text-align: center;
  font-weight: var(--title-weight);
  font-size: var(--title-size);
  color: var(--c-text);
  font-style: italic;
}

.btnUnRegister {
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: var(--b-space-2xl);
}

.phrase {
  font-weight: var(--font-weight-bold);
}
