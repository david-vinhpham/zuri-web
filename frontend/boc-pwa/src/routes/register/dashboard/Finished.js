import React from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import moment from 'moment';
import { string, func, boolean } from 'prop-types';
import noop from 'lodash/noop';
import { Button } from 'homecredit-ui';

import s from './Finished.css';

function Finished(props) {
  const { createAt, showUnRegister, unRegister } = props;
  const createAtMoment = moment(createAt);

  return (
    <React.Fragment>
      <div className={s.finished}>
        <p className={s.phrase}>
          Bạn đã thực hiện đăng ký thiết bị vào lúc{' '}
          {createAtMoment.format('HH:mm')} ngày{' '}
          {createAtMoment.format('DD-MM-YYYY')}.
        </p>
        <p className={s.phrase}>Xin Cảm ơn!</p>
      </div>
      {showUnRegister && (
        <Button onClick={unRegister} className={s.btnUnRegister}>
          Hủy đăng ký
        </Button>
      )}
    </React.Fragment>
  );
}

Finished.defaultProps = {
  createAt: new Date(),
  unRegister: noop,
  showUnRegister: false,
};

Finished.propTypes = {
  createAt: string,
  unRegister: func,
  showUnRegister: boolean,
};

export default withStyles(s)(Finished);
