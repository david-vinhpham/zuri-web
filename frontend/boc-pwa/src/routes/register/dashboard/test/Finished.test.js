import React from 'react';
import renderer from 'react-test-renderer';

import Finished from '../Finished';

it('Finished snapshot', () => {
  const wrapper = renderer.create(
    <Finished createAt="" unRegister={() => {}} showUnRegister />,
  );
  const snapshot = wrapper.toJSON();

  expect(snapshot).toMatchSnapshot();
});
