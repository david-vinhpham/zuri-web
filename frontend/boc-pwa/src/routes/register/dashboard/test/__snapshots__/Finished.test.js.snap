// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finished snapshot 1`] = `
Array [
  <div
    className="finished"
  >
    <p
      className="phrase"
    >
      Bạn đã thực hiện đăng ký thiết bị vào lúc
       
      Invalid date
       ngày
       
      Invalid date
      .
    </p>
    <p
      className="phrase"
    >
      Xin Cảm ơn!
    </p>
  </div>,
  <button
    className="Button_buttonReset__3cQTl Button_button__2pQRN btnUnRegister"
    disabled={false}
    onClick={[Function]}
    title=""
    type="button"
  >
    Hủy đăng ký
  </button>,
]
`;
